### PR TITLE
Feature/model 1

### PR DESCRIPTION
**Pull request by bhylak**
Created the model from this spec: https://www.lucidchart.com/documents/edit/9f86c8e3-c304-43fc-af7e-ff66c22e6c8e

### CODE REVIEW

**Comment by bhylak** [comment]
Also added folder structure.

**Comment by bhylak** [comment]
it'd be great to have this reviewed tonight.

File: app/src/androidTest/java/org/fundacionparaguaya/advisorapp/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -23,6 +23,6 @@ public void useAppContext() throws Exception
         // Context of the app under test.
         Context appContext = InstrumentationRegistry.getTargetContext();
 
-        assertEquals("com.fundacionparaguaya.advisorapp", appContext.getPackageName());
+        assertEquals("com.fundacionparaguaya.org.fundacionparaguaya.advisorapp", appContext.getPackageName());
```
Comment (krconv) [inline app/src/androidTest/java/org/fundacionparaguaya/advisorapp/ExampleInstrumentedTest.java]: I think that the package name got duplicated here; `assertEquals("org.fundacionparaguaya.advisorapp"...`

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
             android:roundIcon="@mipmap/ic_launcher_round"
             android:supportsRtl="true"
             android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name="sorg.fundacionparaguaya.advisorapp.MainActivity">
```
Comment (krconv) [inline app/src/main/AndroidManifest.xml]: Typo in the activity name; `"sorg.fund...` -> `"org.fund...`

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
         xmlns:app="http://schemas.android.com/apk/res-auto"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        tools:context="com.fundacionparaguaya.advisorapp.MainActivity">
+        tools:context="com.fundacionparaguaya.advisorapp.org.fundacionparaguaya.advisorapp.MainActivity">
```
Comment (krconv) [inline app/src/main/res/layout/activity_main.xml]: Think the package name got duplicated here too

File: app/src/main/java/org/fundacionparaguaya/advisorapp/dashboard/TabView.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ * Created by benjaminhylak on 1/11/18.
+ */
+
+public class TabView extends LinearLayoutCompat
+{
+    public TabView(Context context)
+    {
+        super(context);
+    }
+}
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/dashboard/TabView.java]: A comment on this class might help to distinguish what it's purpose is

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (bhylak) [reply app/src/main/res/layout/activity_main.xml]: Good catch. I'll set up an emulator in circleci soon to catch these types of issues... right now, we're only checking compilation issues.

**Comment by bhylak** [comment]
@krconv ~~on folder structure. I mentioned this during scrum today... you're correct that some people break their model, views, controllers (or model view presenters, or model view and view models, etc.) into their own folders. In Android specifically, some people create folders for activities, fragments, etc. 

Personally, I don't really understand the desire to structure it as such. It just makes files hard to find... I think grouping related files makes more sense. If you're working on a fragment file, you would probably want to easily find the view model for the said fragment. 

Here's an example of what I mean: https://github.com/google/iosched/tree/master/lib/src/main/java/com/google/samples/apps/iosched/schedule~~

Looked through the source code in some of [these apps](https://blog.aritraroy.in/20-awesome-open-source-android-apps-to-boost-your-development-skills-b62832cf0fa4)... They all break it up [as you described. ](https://github.com/naman14/Timber/tree/master/app/src/main/java/com/naman14/timber) 

New Proposed folder structure:

- activities
- adapters
- fragments
- util
- viewmodels
- views (view components)

**Comment by krconv** [comment]
Why break it up into the different categories, and not just keep the MVVM folders? I think we've also talked about having different specializations in different areas of the app, and it seems that activities, fragments and views are all more related; also, where do you think models should go?

**Comment by bhylak** [comment]
forgot the models folder -- edited prev comment

Take a look at some of those open source apps I linked to. Every single one structures it like that. 
My previous argument was to keep everything related together. If we're going to break the view models out as you suggested, we should break everything out to keep it consistent.

**Comment by krconv** [comment]
Ok; 👍

**Comment by bhylak** [comment]
- activities 
   -(self descriptive)
- adapters
   -(for lists/recycler view)
- data
   -Classes related to local storage (Room, DAOs, etc.)
- models
- fragments
- sync
  -Classes related to rest/retrofit, authentication, repository
- util
  -Any util classes
- viewmodels
- viewcomponents
  -classes that have the view class as a parent

**Pull request merged**